Dockerfile

```diff
@@ -18,9 +18,17 @@ RUN adduser --disabled-password --gecos "" mitodl
 RUN mkdir /var/media && chown -R mitodl:mitodl /var/media
 
 # Install project packages
-COPY requirements.txt /tmp/requirements.txt
-COPY test_requirements.txt /tmp/test_requirements.txt
-RUN pip install -r requirements.txt -r test_requirements.txt
+ENV  \
+  # poetry:
+  POETRY_VERSION=1.5.1 \
+  POETRY_VIRTUALENVS_CREATE=false \
```

Comment:
> Keep the Poetry virtualenv disabled, and add / remove dependencies with docker compose exec web poetry add / remove foo

That actually doesn't work because poetry is editing system python, which mitodl user does not have permission to do. You can:
1. `docker compose exec -u root web poetry add/remove <package-name>
2. OR `poetry add/remove --lock <packagename>`, either inside or outside of docker, then rebuild the container.

Initially (2) seemed annoying to me. But in practice I don't think we add/remove packages that often. Though I suppose it would be annoying if you were working on a branch and constantly needing to add/remove packages to try different versions or something.

Reply:
> That actually doesn't work because poetry is editing system python, which mitodl user does not have permission to do. You can:

Yea, I meant to indicate the `-u root` in my comment but forgot, that was my intention. In the case where you're trying out different package versions, another option would be to simply open a terminal and `docker compose exec -u root web /bin/bash` and then just have a terminal open where you can play with that.

I don't have any super strong opinions on this aside from thinking we should probably not disable the virtualenv because it seems to be not recommended.

Reply:
@ChristopherChudzicki I think whatever we do, we should add a small section to the README to indicate how dependencies should be added / removed

Reply:
> I think whatever we do, we should add a small section to the README to indicate how dependencies should be added / removed

I did **not** do this because:
- I don't think we should have a section like that in every repo. I guess it could go in https://github.com/mitodl/handbook/blob/master/how-to/common-web-app-guide.md, but better of just reading the poetry docs 
- and with the virtual env, the commands are just poetry add/remove etc